libs/source/Source.mli

```diff
@@ -0,0 +1,60 @@
+(* Cooper Pierce
+ *
+ * Copyright (c) Semgrep Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(** Sources of content which may be consumed by Semgrep.
+
+   A source is the {e user-relevant location} where content (e.g., source code)
+   originated. This should be used when storing a location which will later be
+   reported to a user.
```

Comment:
Then what is the difference between a source and a Tok.t (or Pos.t, or Loc.t).

Reply:
This is "location" in a more macro sense. E.g., a specific file name; standard input; a specific network-accessible uri; a specific git object which designates a file present at a particular commit.

I will add a further note to differentiate.

Reply:
Done in 194e4c0c5fc4a4ac1fab02432f90a9cd3a75d8c9.